jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -159,6 +159,17 @@ typedef enum
 #define ECMA_ASSERT_VALUE_IS_SYMBOL(value) (false)
 #endif /* ENABLED (JERRY_ES2015) */
 
+/**
+ * Check whether the given object has [[ProxyHandler]] and [[ProxyTarger]] internal slots
+ *
+ * @param obj_p ecma-object
+ */
+#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
+#define ECMA_OBJECT_IS_PROXY(obj_p) (JERRY_UNLIKELY (ecma_get_object_type ((obj_p)) == ECMA_OBJECT_TYPE_PROXY))
+#else /* !ENABLED (JERRY_ES2015_BUILTIN_PROXY) */
+#define ECMA_OBJECT_IS_PROXY(obj_p) (false)
```

Comment:
Is there a code where this is used?

Reply:
Yes, in ordinary-object methods I use this macro to assert that the object cannot be proxy.